File: lua/source.lua
function minifyLua(code)
    local strippedCode = string.gsub(code, "%s+", " ")
    local minifiedCode = string.gsub(strippedCode, "%s*([%(%){}%[%],;])%s*", "%1")
    minifiedCode = string.gsub(minifiedCode, ";(%s*})", "%1")
    return minifiedCode
end

function unminifyLua(code)
    local formattedCode = string.gsub(code, ";", ";\n")
    formattedCode = string.gsub(formattedCode, "([^%w])function", "%1\nfunction")
    formattedCode = string.gsub(formattedCode, "end([^%w])", "end\n%1")
    formattedCode = string.gsub(formattedCode, "([=%+%-%*/%%])", " %1 ")
    return formattedCode
end

File: java/source.java
public class LuaMinifier {
    public static String minifyLua(String code) {
        String strippedCode = code.replaceAll("\\s+", " ");
        String minifiedCode = strippedCode.replaceAll("\\s*([\\(\\)\\{\\}\\[\\],;])\\s*", "$1");
        minifiedCode = minifiedCode.replaceAll(";(\\s*\\})", "$1");
        return minifiedCode;
    }

    public static String unminifyLua(String code) {
        String formattedCode = code.replaceAll(";", ";\n");
        formattedCode = formattedCode.replaceAll("([^\\w])function", "$1\nfunction");
        formattedCode = formattedCode.replaceAll("end([^\\w])", "end\n$1");
        formattedCode = formattedCode.replaceAll("([=%+\\-*/%])", " $1 ");
        return formattedCode;
    }
}